packages/next/build/webpack-config.ts

```diff
@@ -2182,3 +2183,18 @@ export default async function getBaseWebpackConfig(
 
   return webpackConfig
 }
+
+function errorIfEnvConflicted(
+  config: NextConfigComplete,
+  key: string,
+  isServer: boolean
+) {
+  const isPrivateKey = /^(?:NODE_.+)|^(?:__.+)$/i.test(key)
+  const hasNextRuntimeKey = isServer && key === 'NEXT_RUNTIME'
```

Comment:
```suggestion
  const hasNextRuntimeKey = key === 'NEXT_RUNTIME'
```
Should we error if it's set for client build or server build for consistency?